### PR TITLE
Code Comments

### PR DESCRIPTION
**Pull request by dino-rodriguez**
**Context:** On the community call, we chatted about code comments and how they could be helpful in the interledger-js code.

I made some comments on the BTP code as I was going through it some time ago. They might be too verbose -- not sure. The comments are tsdocs compatible, and use some markdown styling as well for the generated documentation.

Not sure if this is something that should be merged into master or not, but thought it would be worth posting for consideration. Thanks @adrianhopebailie for bringing that up on the call.

### CODE REVIEW

**Comment by karzak** [comment]
Thanks @d1no007! +1 on having commented code in master.

File: tsconfig.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "inlineSources": true,
     "strictNullChecks": true,
     "suppressImplicitAnyIndexErrors": true,
+    "allowUnreachableCode": true
```
Comment (kincaidoneil) [inline tsconfig.json]: Is this necessary? If not, it'd probably be best to leave it off

**Review by kincaidoneil — APPROVED** [review]
This looks great to me; very thorough, and would certainly be helpful for plugin authors 👍

**Comment by dino-rodriguez** [comment]
Great -- thanks @kincaidoneil @karzak! Seems a break statement after a return within a switch statement was causing the unreachable code. Changed that and turned the compiler option off.

**Review by adrianhopebailie — APPROVED** [review]
LGTM, we should look at auto-publishing the API docs as part of each release

**Pull request merged**